library/date/iso8601_spec.rb

```diff
@@ -0,0 +1,38 @@
+require_relative '../../spec_helper'
+require 'date'
+
+describe "Date.iso8601" do
+  it "parses YYYY-MM-DD into a Date object" do
+    d = Date.iso8601("2018-01-01")
+    d.should == Date.civil(2018, 1, 1)
+  end
+
+  it "parses YYYYMMDD into a Date object" do
+    d = Date.iso8601("20180715")
+    d.should == Date.civil(2018, 7, 15)
+  end
+
+  it "parses a negative Date" do
+    d = Date.iso8601("-4712-01-01")
+    d.should == Date.civil(-4712, 1, 1)
+  end
+
+  it "parses a Symbol into a Date object" do
+    d = Date.iso8601(:'2015-10-15')
+    d.should == Date.civil(2015, 10, 15)
+  end
+
+  it "parses a StringSubclass into a Date object" do
+    StringSubclass = Class.new(String)
```

Comment:
This still assigns a constant at the top-level, can you make it a local variable instead?
Alternatively, you could also use mocks like `object = mock("str")`, `object.should_receive(:to_str).and_return("-4712-01-01")`, but then it doesn't really test a subclass instance, it's more general.

Reply:
I should probably add a check in CI that no unexpected top-level constant is added.
Might be tricky to implement though.